src/alerts/time_series.py

```diff
@@ -8,24 +8,150 @@
 import matplotlib.pyplot as plt
 import pandas as pd
 from alerts.anomaly import AnomalyDetector
+from alerts.dataclasses import TimeSeries, PredictionInterval, Event, LimitInterval
 
+from abc import abstractmethod
+from abc import ABC
 
-class AnomalyTS():
 
-    QUANTILE = 1.625
+def identity(x: Any) -> Any:
+    return x
+
+
+class AnomalyTS(ABC):
+
+    @abstractmethod
+    def prediction_interval() -> List[PredictionInterval]:
+        pass
+
+    @abstractmethod
+    def plot(self) -> None:
+        pass
+
+
+class AnomalyMeanTS(AnomalyTS):
+
+    Z_SCORE = 1.625
     WINDOW = 30
 
     def __init__(self,
-                 T: List[Tuple[Any, int]],
-                 parameters: Dict[str, Any]):
-        """Time series anomaly
-
-        Args:
-            T (List[Tuple[int, int]]): represents data point containing a
-        timestamp and an associating value.
-            parameters (Dict[str, Any]): set of hyper-parameters that help to
-                compute the prediction intervals.
+                 ts: TimeSeries,
+                 z_score: float = Z_SCORE,
+                 window: int = WINDOW,
+                 flat_ts: bool = False):
+        self.z_score = z_score
+        self.window = window
+        self.flat_ts = flat_ts
+        self.ts = ts
```

Comment:
Why not applying flat here instead of checking in each case the code need? Have sense?

```suggestion
        self.ts = ts
        self.ts.values =  np.sqrt(ts.values) if self.flat_ts else ts.values
```

Reply:
@quimdt It has sense yes